src/resource/ResourceManager.cpp

```diff
@@ -410,4 +410,10 @@ void ResourceManager::SetAltAssetsEnabled(bool isEnabled) {
     mAltAssetsEnabled = isEnabled;
 }
 
+void ResourceManager::ShutDownThreadPool() {
+    mThreadPool->pause();
+    mThreadPool->wait_for(std::chrono::duration<double>(2));
```

Comment:
i see you mentioned
> I have a wait time of 2 seconds here between pausing and purging to prevent deadlocks

that feels a little bit magic to me, are we just assuming we'll never have a pool where 2 seconds isn't enough time? have you looked into alternative ways to mitigate deadlock that don't rely on a hardcoded delay?

Reply:
@Kenix3's is to just pass the amount of time to wait as a function param